From bc48d30fa977269af13b26f053806fe164f86be3 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Wed, 24 Oct 2018 15:21:41 +0800
Subject: [PATCH 4945/5242] MLK-20045-2 usb: chipidea: usbmisc_imx: delete the
 HSIC operation for imx7d

commit  a8f37d55a724413c7044d67f84a3ab6e1d635646 from
https://source.codeaurora.org/external/imx/linux-imx.git

For imx7d, no software operation is needed for HSIC controller, so delete
related operation.

Reviewed-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/usbmisc_imx.c |   11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/usb/chipidea/usbmisc_imx.c b/drivers/usb/chipidea/usbmisc_imx.c
index ae45a0b..558ac44 100644
--- a/drivers/usb/chipidea/usbmisc_imx.c
+++ b/drivers/usb/chipidea/usbmisc_imx.c
@@ -647,14 +647,7 @@ static int usbmisc_imx7d_init(struct imx_usbmisc_data *data)
 	reg = readl(usbmisc->base);
 	writel(reg | MX6_BM_NON_BURST_SETTING, usbmisc->base);
 
-	if (data->hsic) {
-		reg = readl(usbmisc->base);
-		writel(reg | MX6_BM_UTMI_ON_CLOCK, usbmisc->base);
-
-		reg = readl(usbmisc->base + MX6_USB_HSIC_CTRL_OFFSET);
-		reg |= MX6_BM_HSIC_EN | MX6_BM_HSIC_CLK_ON;
-		writel(reg, usbmisc->base + MX6_USB_HSIC_CTRL_OFFSET);
-	} else {
+	if (!data->hsic) {
 		reg = readl(usbmisc->base + MX7D_USBNC_USB_CTRL2);
 		reg &= ~MX7D_USB_VBUS_WAKEUP_SOURCE_MASK;
 		writel(reg | MX7D_USB_VBUS_WAKEUP_SOURCE_BVALID
@@ -1023,8 +1016,6 @@ static int usbmisc_imx7ulp_init(struct imx_usbmisc_data *data)
 	.power_lost_check = usbmisc_imx7d_power_lost_check,
 	.charger_detection = imx7d_charger_detection,
 	.term_select_override = usbmisc_term_select_override,
-	.hsic_set_connect = usbmisc_imx6_hsic_set_connect,
-	.hsic_set_clk   = usbmisc_imx6_hsic_set_clk,
 };
 
 static const struct usbmisc_ops imx7ulp_usbmisc_ops = {
-- 
1.7.9.5

